src/Microsoft.TestPlatform.AdapterUtilities/HierarchyConstants.cs

```diff
@@ -0,0 +1,52 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.TestPlatform.AdapterUtilities
+{
+    /// <summary>
+    /// Contants to help declare Hierarchy test property.
+    /// </summary>
+    public static class HierarchyConstants
+    {
+        /// <summary>
+        /// Label to use on Hierarchy test property.
+        /// </summary>
+        public const string HierarchyLabel = "Hierarchy";
+
+        /// <summary>
+        /// Property id to use on Hierarchy test property.
+        /// </summary>
+        public const string HierarchyPropertyId = "TestCase." + HierarchyLabel;
+
+        /// <summary>
+        /// Meanings of the indices in the Hierarchy array.
+        /// </summary>
+        public static class Levels
```

Comment:
Could the HierarchyName property be a dictionary instead, so both setting/retrieving is easier to understand and uniqueness of values is guaranteed? We could serialize/deserialise this the way we want internally,

Reply:
Do you mean like a `<string, string>` directory with set keys on the adapter?

Reply:
Gosh, I dont have a good way of responding to notifications from github. Yes, I meant a <string,string> property bag.